app/controllers/ems_common.rb

```diff
@@ -578,6 +579,18 @@ def recheck_authentication(id = nil)
     @record.authentication_check_types_queue(@record.authentication_for_summary.pluck(:authtype), :save => true)
   end
 
+  def check_compliance(model)
+    showlist = @lastaction == "show_list"
+    emss = find_checked_items
+    if emss.empty?
+      add_flash(_("No %{record} were selected for %{task}") % {model => ui_lookup(:models => model),
+                                                               :task  => "Compliance Check"}, :error)
+    end
+    process_emss(emss, "check_compliance")
+    showlist ? show_list : show
```

Comment:
@alongoldboim dont need to set showlist variable as it is not being used anywhere else, you could just do 
```
@lastaction == "show_list" ? show_list : show
```